src/crypto/mod.rs

```diff
@@ -49,9 +59,10 @@ pub fn generate_pseudorandom_bytes(
 /// Compute keyed hmac
 pub fn compute_keyed_hmac<D>(key: &[u8], data: &[u8]) -> HmacOutput<D>
 where
-    D: Update + BlockInput + FixedOutput + Reset + Default + Clone,
-    D::BlockSize: ArrayLength<u8>,
-    D::OutputSize: ArrayLength<u8>,
+    D: CoreProxy,
+    D::Core: HashMarker + FixedOutputCore + BufferKindUser<BufferKind = Eager> + Default + Clone,
+    <D::Core as BlockSizeUser>::BlockSize: IsLess<U256>,
+    Le<<D::Core as BlockSizeUser>::BlockSize, U256>: NonZero,
```

Comment:
those seem like extremely weird type bounds, are you sure we actually need them?

Reply:
They look definitely weird, I'm not sure about them. I tried different combination, but couldn't get it to work. Type bounds changed a lot. Could you have a look?